internal/handlers/common/compare.go

```diff
@@ -188,15 +189,24 @@ func compare(docValue, filter any) compareResult {
 
 	switch docValue := docValue.(type) {
 	case *types.Document:
+		filter, isFilterDocument := filter.(*types.Document)
+		if isFilterDocument {
+			return compareDocuments(filter, docValue)
+		}
 		return notEqual
 
 	case *types.Array:
+		switch filter.(type) {
```

Comment:
```suggestion
		switch filter := filter.(type) {
```

Then `filter` can be used without type assertion below.

But then why we use different type checking style there and for document above? Please use the same, which is more common in that file

Reply:
Noted and fixed